addons/account/models/account_move_line.py

```diff
@@ -1309,18 +1320,6 @@ def write(self, vals):
             if any(self.env['account.move']._field_will_change(line, vals, field_name) for field_name in protected_fields['reconciliation']):
                 line._check_reconciliation()
 
-            # Check switching receivable / payable accounts.
-            if account_to_write:
-                account_type = line.account_id.account_type
-                if line.move_id.is_sale_document(include_receipts=True):
-                    if (account_type == 'asset_receivable' and account_to_write.account_type != account_type) \
-                            or (account_type != 'asset_receivable' and account_to_write.account_type == 'asset_receivable'):
-                        raise UserError(_("You can only set an account having the receivable type on payment terms lines for customer invoice."))
-                if line.move_id.is_purchase_document(include_receipts=True):
-                    if (account_type == 'liability_payable' and account_to_write.account_type != account_type) \
-                            or (account_type != 'liability_payable' and account_to_write.account_type == 'liability_payable'):
-                        raise UserError(_("You can only set an account having the payable type on payment terms lines for vendor bill."))
```

Comment:
I'm not sure why you need to remove these checks... Is there something else preventing you to change a receivable to a payable in a customer invoice?